csv_detective/utils.py

```diff
@@ -58,19 +59,26 @@ def test_col(table, all_tests, num_rows, output_mode):
     test_funcs = dict()
     for test in all_tests:
         name = test.__name__.split('.')[-1]
-
         test_funcs[name] = {
             'func': test._is,
             'prop': test.PROPORTION
         }
-
     return_table = pd.DataFrame(columns=table.columns)
     for key, value in test_funcs.items():
+        # When analysis of all file is requested (num_rows = -1) we fix a threshold of 
+        # 1000 rows for every checks outside int or float format
+        local_num_rows = max(num_rows, 1000)
```

Comment:
If user inputs a value between 1 and 1000 for a column that is not in the list below, doesn't this force a larger analysis than requested?

Reply:
hum, I don't think so. If num_rows = 500 for instance, every column will be analyzed for first 500 rows except for columns below where whole column will be analyzed. If num_rows > 1000 we will analyzed only first 1000 rows except for int or float checks

Reply:
ok my bad I understand I will correct it